spec/unit/indirector/rest_spec.rb

```diff
@@ -353,7 +353,7 @@ def save_request(key, instance, options={})
       it "calls get on the connection" do
         request = find_request('foo bar')
 
-        connection.expects(:get).with("#{url_prefix}/test_model/foo%20bar?environment=production&", anything).returns(mock_response('200', 'response body'))
+        connection.expects(:get).with("#{url_prefix}/test_model/foo+bar?environment=production&", anything).returns(mock_response('200', 'response body'))
```

Comment:
Yeah, so I think in URIs, the <space> / `%20` / `+` *should* be treated equivalently as far as HTTP goes.  It's more of a concern when strings get parsed afterwards and used as string data - think hash keys and other string comparisons.

Looks like I'll have to do a little bit of investigation into the Jetty code or similar to see how URIs are canonicalized - but it does look based on your logs that the `+` is passed through, which is probably undesirable for backwards compat.